nixos/modules/system/boot/loader/systemd-boot/systemd-boot-builder.py

```diff
@@ -65,7 +65,7 @@ def from_path(cls: Type["Entry"], path: Path) -> "Entry":
         # Matching nixos*-generation-$number*.conf
         rex_generation = re.compile(r"^nixos.*-generation-([0-9]+).*\.conf$")
         # Matching nixos*-generation-$number-specialisation-$specialisation_name*.conf
-        rex_specialisation = re.compile(r"^nixos.*-generation-([0-9]+)-specialisation-([a-zA-Z0-9]+).*\.conf$")
+        rex_specialisation = re.compile(r"^nixos.*-generation-([0-9]+)-specialisation-(.*)\.conf$")
```

Comment:
So, while I believe this does fix the bug, the problem is that I *think* it's going to break boot counting. With boot counting, there's an additional `+$countdown-$tries` before the `.conf` until the boot succeeds, at which point that suffix is removed.

Reply:
Oh I see, yeah this naive change I think would break it, but I can't be sure either since I haven't tested it with boot counting nor am I familiar with that new change. I didn't notice that the commit which introduced this new regex was more restrictive when matching the specialisation's name on purpose, makes more sense now.

I will defer to other's thoughts and final opinions for this change, as I am a bit less informed.